### PR TITLE
[RFC] eval,fileio: Omit additional fsync() call

### LINKED ISSUE #6420: write_list: Slowdown in Docker because of fsync
I have noticed a slowdown when running Neovim in Docker.

It turned out that https://github.com/neovim/neovim/commit/31cdb227ba60e8867b3cb8e60ae8215290dd85a4#commitcomment-21508188 is causing it.

Reverting that commit on master fixes the regression.

I've used the following method to test it:
```vim
function! Bisect()
  let start = reltimefloat(reltime())
  let file = tempname()
  for i in range(1, 50)
    call writefile([i], file, 'a')
  endfor
  let duration = reltimefloat(reltime()) - start
  echoerr printf('duration: %.3f', duration)
endfunction
```

The time difference is around 0.1s for those 50 runs: 0.352 vs 0.445

(This seems not too much, but it somehow caused Neomake's test suite too take twice as long, since it does a lot of logging.  Although I am not sure about how I've noticed this, since normally logging to a file is not enabled during the tests, but only one test is using it.)

@ZyX-I mentioned in https://github.com/neovim/neovim/commit/31cdb227ba60e8867b3cb8e60ae8215290dd85a4#commitcomment-21508188 that the fsync would be done when freeing the fp, and maybe that the issue/diff here after all: previously fsync would be done just once, while its done now after each write?!

Docker info:
```
Server Version: 17.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Kernel Version: 4.10.6-1-ARCH
Operating System: Arch Linux
OSType: linux
Architecture: x86_64
```

### PR DESCRIPTION
**Pull request by ZyX-I**
Fixes #6420

### CODE REVIEW

File: src/nvim/eval.c [hunk 1]
```diff
@@ -17289,7 +17289,7 @@ static bool write_list(FileDescriptor *const fp, const list_T *const list,
       }
     }
   }
-  if ((error = file_fsync(fp)) != 0) {
+  if ((error = file_flush(fp)) != 0) {
```
Comment (oni-link) [inline src/nvim/eval.c]: Does this still guarantees that the file is really written?

Comment (ZyX-I) [inline src/nvim/eval.c]: This guarantees that buffer contents will be sent to the OS (write() syscall) and buffer will be cleared, nothing more.

Comment (ZyX-I) [inline src/nvim/eval.c]: What is left then is fsync() and close() done on `file_free()`.

**Comment by ZyX-I** [comment]
BTW, why I am allocating file when I can have it on stack?

**Pull request merged**